include/triton/core/tritonserver.h

```diff
@@ -1883,6 +1883,31 @@ TRITONSERVER_DECLSPEC TRITONSERVER_Error* TRITONSERVER_ServerDelete(
 TRITONSERVER_DECLSPEC TRITONSERVER_Error* TRITONSERVER_ServerStop(
     TRITONSERVER_Server* server);
 
+/// Register a new model repository
+///
+/// \param server The inference server object.
+/// \param repository_path The full path to the model repository.
+/// \param name_mapping List of name_mapping parameters. Each mapping has
+/// the model directory name as its key, overriden model name as its value.
+/// If the model directory name is empty, there must only be one model directory
```

Comment:
If the map doesn't contain an entry for a subdir or the model repo, then the model name used is just the name of the subdir... that is how our current model repository works. So the map can be empty or can list only some of the model subdirs... but there is no requirement around having only 1 model subdirectory.

Reply:
I think for SageMaker, the endpoint doesn't know the file structure under the repository to be added, so they don't know what the subdir name is while the endpoint still want to use different name for the model.

Reply:
I can try to word this better. This is specifically for the case we are trying to support. If the user does not know the name of the model repository nor the name of the model, they can still use the model with triton.

They would register a model repository with that single model. The parameter would have an empty directory name a given model name to map it to. Then, they could use that model name with Triton (e.g. to load, run, or unload it).

Reply:
It is not that the user does not know the name of the model. In the case of sagemaker they will be registering model repositories that have only a single model, and the subdir name for that model will always be 'model'. But the SM load-model API allows the user to assign a name to the model and that name is what triton should use. So for SM they will always call the register API with map { 'model' -> '<name user specifies in SM load model API>' }. But we should not need to do anything special for SM... they will just use the generic capability we are designing into the register API.

Reply:
Thanks for clarifying. Removed this edge case based on our conversation.